compiler/rustc_lint/src/unused.rs

```diff
@@ -805,6 +805,18 @@ impl EarlyLintPass for UnusedParens {
                 }
                 return;
             }
+            ExprKind::Match(ref _expr, ref arm) => {
+                for a in arm {
+                    self.check_unused_delims_expr(
+                        cx,
+                        &a.body,
+                        UnusedDelimsCtx::BlockRetValue,
```

Comment:
It would be better to create a new `UnusedDelimsCtx` variant instead of reusing `BlockRetValue`, it will make the warning message more correct.

Reply:
Sure, would the following be the msg?

`unnecessary parentheses around match block`

Im pretty sure there is a more correct way of saying `match block`

Reply:
"`match` arm expression", for example.